nova/virt/vmwareapi/vm_util.py

```diff
@@ -853,14 +853,22 @@ def _get_device_disk_type(device):
             return constants.DEFAULT_DISK_TYPE
 
 
+def _hardware_devices(hardware_devices):
```

Comment:
What do we need this split off for?

Reply:
It isn't really needed, it is the result of iterative refactoring. I first replaced all occurrences matching the code of `_hardware_devices`, and consolidated the `get_object_property(... "config.hardware.device")` calls.
After that, I removed all `_hardware_devices`, which got their output from `hardware_devices`.
And in the end only one call was left.